source/features/copy-file.tsx

```diff
@@ -2,25 +2,65 @@ import React from 'dom-chef';
 import select from 'select-dom';
 import copyToClipboard from 'copy-text-to-clipboard';
 import features from '../libs/features';
+import {fetchSource} from './view-markdown-source';
+
+// Copy to clipboard can't be async, and must be event driven
+let markdown: Element;
+let loaded = false;
+let loading: Promise<any>;
+const isMarkDown = features.isMarkDown();
+
+async function loadFile(): Promise<void> {
+	loaded = true;
+	console.log('loading');
+	loading = new Promise((resolve => {
+		resolve(fetchSource());
+	}));
```

Comment:
`fetchSource` already returns a promise.

Also this globals-based approach means that if you open `file1.md` and then `file2.md`, the globals will still point to `file1.md`

In `view-markdown-source` I avoided this problem with a WeakMap, so each source is connected to its file (actually, the file's button)

Thankfully we don't need to handle any of this: we can only show the `copy` button when the source is shown, just like it happens for [SVG files](https://github.com/sindresorhus/refined-github/blob/master/media/icon.svg). This means you don't have to call `fetchSource` at all, but rather you can call `.dispatchEvent(new CustomEvent('rgh:view-markdown-source'))` and have `copy-file` listen to it. It's much easier